pkg/tracing/tracing.go

```diff
@@ -20,113 +20,354 @@ import (
 func (s *Span) AddEvent(name string, attributes ...attribute.KeyValue) {
-	s.otelSpan.AddEvent(name, trace.WithAttributes(attributes...))
+	if s.otelSpan != nil {
+		s.otelSpan.AddEvent(name, trace.WithAttributes(attributes...))
+	}
+	if s.enhancedSpan != nil {
+		enhancedAttrs := convertToEnhancedAttributes(attributes)
+		s.enhancedSpan.AddEvent(name, enhancedAttrs...)
+	}
 }
 
 // RecordError will record err as an exception span event for this span
 func (s *Span) RecordError(err error, options ...trace.EventOption) {
-	s.otelSpan.RecordError(err, options...)
+	if s.otelSpan != nil {
+		s.otelSpan.RecordError(err, options...)
+	}
+	if s.enhancedSpan != nil {
+		s.enhancedSpan.RecordError(err)
+	}
 }
 
-// SetStatus sets the status of the current span.
-// If an error is encountered, it records the error and sets span status to Error.
+// SetStatus sets the status of both OTEL and enhanced spans.
 func (s *Span) SetStatus(err error) {
-	if err != nil {
-		s.otelSpan.RecordError(err)
-		s.otelSpan.SetStatus(codes.Error, err.Error())
-	} else {
-		s.otelSpan.SetStatus(codes.Ok, "")
+	if s.otelSpan != nil {
+		if err != nil {
+			s.otelSpan.RecordError(err)
+			s.otelSpan.SetStatus(codes.Error, err.Error())
+		} else {
+			s.otelSpan.SetStatus(codes.Ok, "")
+		}
+	}
+	if s.enhancedSpan != nil {
+		if err != nil {
+			s.enhancedSpan.SetStatus(enhanced.StatusCodeError, err.Error())
+			s.enhancedSpan.RecordError(err)
+		} else {
+			s.enhancedSpan.SetStatus(enhanced.StatusCodeOk, "success")
+		}
 	}
 }
 
-// SetAttributes sets kv as attributes of the span.
+// SetAttributes sets attributes on both OTEL and enhanced spans.
 func (s *Span) SetAttributes(kv ...attribute.KeyValue) {
-	s.otelSpan.SetAttributes(kv...)
+	if s.otelSpan != nil {
+		s.otelSpan.SetAttributes(kv...)
+	}
+	if s.enhancedSpan != nil {
+		enhancedAttrs := convertToEnhancedAttributes(kv)
+		s.enhancedSpan.SetAttributes(enhancedAttrs...)
+	}
+}
+
+// SetAttribute sets a single attribute on both OTEL and enhanced spans.
+func (s *Span) SetAttribute(key string, value interface{}) {
+	if s.otelSpan != nil {
+		s.otelSpan.SetAttributes(Attribute(key, value))
+	}
+	if s.enhancedSpan != nil {
+		s.enhancedSpan.SetAttribute(key, value)
+	}
 }
 
 const spanDelimiter = "."
 
-// Name sets the span name by joining a list of strings in dot separated format.
-func Name(names ...string) string {
-	return strings.Join(names, spanDelimiter)
+// Name composes a span operation name using the standard delimiter.
+func Name(parts ...string) string {
+	return strings.Join(parts, spanDelimiter)
 }
 
-// Attribute takes a key value pair and returns attribute.KeyValue type.
+// SpanOperation returns a composite span operation name.
+func SpanOperation(parts ...string) string {
+	if len(parts) == 0 {
+		return ""
+	}
+	return strings.Join(parts, spanDelimiter)
+}
+
+// Attribute returns generic attribute.KeyValue type for any supported input types.
 func Attribute(k string, v any) attribute.KeyValue {
```

Comment:
Just be curiosity, L281 - L294 has been covered by keyValue. Why we still need this additional implementation here?

Reply:
I reused KayValue here